actors/miner/tests/miner_actor_test_construction.rs

```diff
@@ -0,0 +1,163 @@
+        .actor_code_cids
+        .insert(env.owner, *ACCOUNT_ACTOR_CODE_ID);
+    env.rt
+        .actor_code_cids
+        .insert(env.worker, *ACCOUNT_ACTOR_CODE_ID);
+    env.rt
+        .actor_code_cids
+        .insert(env.control_addrs[0], *ACCOUNT_ACTOR_CODE_ID);
+    env.rt
+        .actor_code_cids
+        .insert(env.control_addrs[1], *ACCOUNT_ACTOR_CODE_ID);
+    env.rt.hash_func = Box::new(blake2b_256);
+    env.rt.caller = *INIT_ACTOR_ADDR;
+    env.rt.caller_type = *INIT_ACTOR_CODE_ID;
+    env
+}
+
+fn constructor_params(env: &TestEnv) -> ConstructorParams {
+    ConstructorParams {
+        owner: env.owner,
+        worker: env.worker,
+        control_addresses: env.control_addrs.clone(),
+        window_post_proof_type: RegisteredPoStProof::StackedDRGWindow32GiBV1,
+        peer_id: env.peer_id.clone(),
+        multi_addresses: env.multiaddrs.clone(),
+    }
+}
+
+#[test]
+fn simple_construction() {
+    let mut env = prepare_env();
+    let params = constructor_params(&env);
+
+    env.rt.expect_validate_caller_addr(vec![*INIT_ACTOR_ADDR]);
+    env.rt.expect_send(
+        env.worker,
+        AccountMethod::PubkeyAddress as u64,
+        RawBytes::default(),
+        TokenAmount::from(0),
+        RawBytes::serialize(env.worker_key).unwrap(),
+        ExitCode::Ok,
+    );
+
+    let result = env
+        .rt
+        .call::<Actor>(
+            Method::Constructor as u64,
+            &RawBytes::serialize(params).unwrap(),
+        )
+        .unwrap();
+    assert_eq!(result.bytes().len(), 0);
+
+    let state = env.rt.get_state::<State>().unwrap();
+
+    let info = state.get_info(&env.rt.store).unwrap();
+    assert_eq!(env.owner, info.owner);
+    assert_eq!(env.worker, info.worker);
+    assert_eq!(env.control_addrs, info.control_addresses);
+    assert_eq!(env.peer_id, info.peer_id);
+    assert_eq!(env.multiaddrs, info.multi_address);
+    assert_eq!(
+        RegisteredPoStProof::StackedDRGWindow32GiBV1,
+        info.window_post_proof_type
+    );
+    assert_eq!(SectorSize::_32GiB, info.sector_size);
+    assert_eq!(2349, info.window_post_partition_sectors);
+
+    assert_eq!(TokenAmount::from(0), state.pre_commit_deposits);
+    assert_eq!(TokenAmount::from(0), state.locked_funds);
+    assert_ne!(Cid::default(), state.pre_committed_sectors);
+    assert_ne!(Cid::default(), state.sectors);
+
+    // according to original specs-actors test, this is set by running the code; magic...
+    let proving_period_start = -2222;
+    assert_eq!(proving_period_start, state.proving_period_start);
+    // this is supposed to be the proving period cron
+    let dl_idx = (env.rt.epoch - proving_period_start) / WPOST_CHALLENGE_WINDOW;
+    assert_eq!(dl_idx as u64, state.current_deadline);
+
+    let deadlines = env
+        .rt
+        .store
+        .get_cbor::<Deadlines>(&state.deadlines)
+        .unwrap()
+        .unwrap();
+    for i in 0..WPOST_PERIOD_DEADLINES {
+        let c = deadlines.due[i as usize];
```

Comment:
You can just iterate over `deadlines.due`. Or `deadlines.due.iter().enumerate()`

Reply:
yeah, I was following the go test and didn't think rust enough.

Reply:
I think i'll keep it braindead to be as close to the original test as possible.